### PR TITLE
Register-RabbitMqEvent fix and minor Features

### LINKED ISSUE #19: Register-RabbitMQEvent not working when not setting Creds as Param
I find out that there was a breaking change in a recent commit here:
https://github.com/RamblingCookieMonster/PSRabbitMq/blob/cbc019644d5282ecafb8244b5e57dc2fe7cf3e8b/PSRabbitMq/Register-RabbitMqEvent.ps1#L140-L152

When Credential is not used as parameter, `$ArgList` is not set.

### PR DESCRIPTION
**Pull request by gaelcolas**
Fixed #19 
Added Parameter -ActionData which will be available from the $ActionSB's scope.
Added Parameter -ListenerJobName defaulting to  `"RabbitMq_${ComputerName}_${Exchange}_${Key}"` for backward compatibility.

### CODE REVIEW

File: PSRabbitMq/Connect-RabbitMqChannel.ps1 [hunk 1]
```diff
@@ -84,7 +92,12 @@
     {
         $Channel = $Connection.CreateModel()
 
-         Write-Progress -id 10 -Activity 'Create SCMB Connection' -Status 'Attempting connection to channel' -PercentComplete 80
+        Write-Progress -id 10 -Activity 'Create SCMB Connection' -Status 'Attempting connection to channel' -PercentComplete 80
+
+        #Actively declare the Exchange (as non-autodelete, non-durable)
+        if($ExchangeType -and [string]::Empty -ne $Exchange) {
```
Comment (gpduck) [inline PSRabbitMq/Connect-RabbitMqChannel.ps1]: Do we need to test for $Exchange -ne $null as well here?  Should we be using:

-not [String]::IsNullOrEmpty($Exchange)

**Comment by gaelcolas** [comment]
You're right, amended.
Thanks!
